javascript/javascript3/week1/cool_api.js

```diff
@@ -0,0 +1,87 @@
+const input = document.getElementById("city");
+const mainDiv = document.getElementById("weather-data");
+let userInput = "";
+function onChange(event) {
+    userInput = event.target.value;
+}
+input.addEventListener("input", onChange);
+
+function showCurrentWeather() {
+    if(userInput === "") {
+        alert("Please type your city name first...");
+    } else {
+        document.getElementById("input-div").style.display = "none";
+        fetch(`https://api.openweathermap.org/data/2.5/weather?q=${userInput}&appid=a1d744b03825e0abb2904f3057933962`)
+        .then(response => response.json())
+        .then(response => {
+            console.log(response);
+            const city = document.createElement("h2");
+            city.innerText = `${response.name}`;
+            mainDiv.appendChild(city);
+            const icon = document.createElement("h4");
+            icon.innerText = `${response.weather[0].main}`;
+            mainDiv.appendChild(icon);
+            const temp = document.createElement("h1");
+            temp.innerText = `${Math.floor(response.main.temp - 273.15)}°`;
+            mainDiv.appendChild(temp);
+
+            const hrElement = document.createElement("hr");
+            mainDiv.appendChild(hrElement);
+
+            const paragraph = document.createElement("p");
+            const description = `Today: ${response.weather[0].description} currently. It's ${temp.innerText}, 
+            the high today was forecast as ${Math.floor(response.main.temp_max - 273.15)}° and 
+            the low today was forecast as ${Math.floor(response.main.temp_min - 273.15)}°.`;
+            paragraph.innerText = description;
+            mainDiv.appendChild(paragraph);
+            const hrElement2 = document.createElement("hr");
+            mainDiv.appendChild(hrElement2);
+
+            const infoDiv = document.createElement("div");
+            mainDiv.appendChild(infoDiv);
+
+            const currentTemp = document.createElement("h3");
+            currentTemp.innerText = `Current temperature: ${temp.innerText}`;
+            infoDiv.appendChild(currentTemp);
+            const feelsLike = document.createElement("h3");
+            feelsLike.innerText = `Feels like: ${Math.floor(response.main.feels_like - 273.15)}°`;
```

Comment:
for example the expression `${Math.floor(x- 273.15)}` is used in many places, which means is is probably a good idea to make a function out of it (with `x` being a parameter)

Reply:
Yes. Will do that.